lib/engine/game/g_1880/game.rb

```diff
@@ -650,6 +664,29 @@ def check_for_foreign_investor_connection(fi)
           graph = Graph.new(self, no_blocking: true)
           graph.reachable_hexes(corporation).include?(fi_home_token)
         end
+
+        def after_par(corporation)
+          super
+          @par_order[corporation.par_price.price] << corporation
+        end
+
+        def operating_order
+          @minors.select(&:floated?) + @par_order.values.flatten
+        end
+
+        def after_buying_train(train)
+          return if train.name == @depot.upcoming.first.name
+
+          @turn += 1
+          @saved_or_round = @round
+          @round = new_stock_round
```

Comment:
Do you have to do the same logic for when a full round happens with no depot trains being bought? If it is in another place, make the round transition logic common.

Reply:
No, when all trains are removed i use depot.export_all which calls @game.phase.buying_train!, which calls after_buying_train, the hook i created in this pr